src/components/CallView/Grid/Grid.vue

```diff
@@ -478,6 +484,73 @@ export default {
 		stripeOpen() {
 			return this.$store.getters.isStripeOpen && !this.isRecording
 		},
+
+		participantsInitialised() {
+			return this.$store.getters.participantsInitialised(this.token)
+		},
+
+		isGuestNonModerator() {
+			return this.$store.getters.getActorType() === ATTENDEE.ACTOR_TYPE.GUESTS
+				&& this.$store.getters.conversation(this.token).participantType !== PARTICIPANT.TYPE.GUEST_MODERATOR
+		},
+
+		orderedVideos() {
+			// Dynamic ordering is not possible for guests because
+			// participants store is not initialized
+			if (this.isGuestNonModerator) {
+				return this.videos
+			}
+
+			if (!this.participantsInitialised) {
+				return []
+			}
+
+			const objectMap = {
+				modelsWithScreenshare: [],
+				modelsTempPromoted: [],
+				modelsWithVideoEnabled: [],
+				modelsWithAudioOnly: [],
+				modelsWithNoPermissions: [],
+			}
+			const screensSet = new Set(this.screens)
+			const videosTempPromotedSet = new Set(this.videosTempPromoted.map(model => model.attributes.nextcloudSessionId))
+			const videoTilesMap = new Map()
+			const audioTilesMap = new Map()
+
+			this.callParticipantModels.forEach((model) => {
+				if (screensSet.has(model.attributes.peerId)) {
+					objectMap.modelsWithScreenshare.push(model)
+				} else if (videosTempPromotedSet.has(model.attributes.nextcloudSessionId)) {
+					objectMap.modelsTempPromoted.push(model)
+				} else if (this.isModelWithVideo(model)) {
+					videoTilesMap.set(model.attributes.nextcloudSessionId, model)
+				} else if (this.isModelWithAudio(model)) {
+					audioTilesMap.set(model.attributes.nextcloudSessionId, model)
+				} else {
+					objectMap.modelsWithNoPermissions.push(model)
+				}
+			})
+
+			objectMap.modelsWithVideoEnabled = this.getOrderedTiles(videoTilesMap, this.tilesWithVideoMask)
+			objectMap.modelsWithAudioOnly = this.getOrderedTiles(audioTilesMap, this.tilesWithAudioMask)
+
+			return [...objectMap.modelsWithScreenshare,
+				...objectMap.modelsTempPromoted,
+				...objectMap.modelsWithVideoEnabled,
+				...objectMap.modelsWithAudioOnly,
+				...objectMap.modelsWithNoPermissions]
+		},
+
+		speakers() {
+			return this.callParticipantModels.filter(model => model.attributes.speaking)
+		},
+
+		speakersWithAudioOff() {
+			if (this.videosTempPromoted.length === 0) {
+				return
```

Comment:
```suggestion
				return []
```

To be consistent with return type.
Another option: drop early return. `[ ].filter()` will return `[ ]` anyway